### ISSUE TITLE
Abaqus wrapper: allow mixed interface

### ISSUE THREAD

**Issue by hdolfen** [pyfsi/coconut#78]
Currently, the Abaqus wrapper cannot handle an interface which has mixed element types (e.g. when using a hexdominant mesh). 

The error starts with the [make_elements function](https://github.com/pyfsi/coconut/blob/78281e4d8e7f4a929058cb70e0e1b88e1ed1d435/coupling_components/solver_wrappers/abaqus/614.py#L554) writing the number of load points of _the last face it has looped over_. This value is read again [later on in the code](https://github.com/pyfsi/coconut/blob/78281e4d8e7f4a929058cb70e0e1b88e1ed1d435/coupling_components/solver_wrappers/abaqus/614.py#L247) as `n_lp`. Eventually something goes wrong [when the input modelpart is created](https://github.com/pyfsi/coconut/blob/78281e4d8e7f4a929058cb70e0e1b88e1ed1d435/coupling_components/solver_wrappers/abaqus/614.py#L263-L278). In the example of mixed triangular and quadrilateral faces, `n_lp` is either 3 or 4. For looping over the faces file the product `n_elem*n_lp` is used as range. But if `n_lp`was put on 3, too little lines of the faces file will be read (as some elements have 4 lines). It is not clear where it would go wrong, maybe a bounding box error would be raised about the interfaces, but it's also possible that the interpolation will just be inadequate. If `n_lp` was put on 4, then the reading of the faces file will be continued too far, raising [this error](https://github.com/pyfsi/coconut/blob/78281e4d8e7f4a929058cb70e0e1b88e1ed1d435/coupling_components/solver_wrappers/abaqus/614.py#L272).

I think this can be solved by changing the way we loop over the faces file.

**Comment by hdolfen** [comment]
@toondm If I'm correct this is the issue Thomas faced just before the Christmas holidays.

**Comment by toondm** [comment]
That's correct @hdolfen. He had a quad-dominant face mesh (i.e. quads with one or a few triangles) and indeed got the error you refer to.

**Comment by awbral** [comment]
I think the problem is fairly easy to solve.

Now the elements file has the following structure:

`n_elem` (total number of elements)
`n_lp` (number of load points per element)
list of element numbers

I suggest to change the `n_lp` on the second line into the _total_ number of load points instead of the number of load points per element. This solves almost every check that could go wrong, as the current `n_lp` is nearly always multiplied with `n_elem`. There is one check that might still fail, namely [this one](https://github.com/pyfsi/coconut/blob/master/coupling_components/solver_wrappers/abaqus/v614.py#L277). Therefore, I suggest to include the number of load points that each individual element has next to the list of element numbers in the elements file (so the elements file gets two columns when loaded into an array). As such, the `n_lp` in this check will be replaced by the number of load points in that particular element (let's call it e.g. `n_lp_loc`).
